app/js/addCategories.ts

```diff
@@ -0,0 +1,46 @@
+async function addCategory():Promise<void> {
+    let categoriesObject = await getData('/category');
+    const responseMessage = document.querySelector('#addedCategoryConfirmation');
+    document.querySelector<HTMLFormElement>('.categoriesForm').addEventListener('submit', (e) => {
+        e.preventDefault();
+        let newCategory: string = document.querySelector<HTMLFormElement>('#categoryName').value;
+        let listOfCategories = [];
+        categoriesObject.data.forEach((category) => {
+            listOfCategories.push(category.name);
+        })
+
+        if (!listOfCategories.includes(newCategory)) {
+
+            if (newCategory.length > 255) {
```

Comment:
I feel like this error could be thrown sooner. Why not immediately check the newCategory length and throw if it's too long rather than doing all the work of forEaching over the returned data, etc if you could know sooner whether to throw? Throw as early as you can to avoid doing unnecessary work